### ISSUE TITLE
Error 403 at Crunchyroll even with cookies (yt-dlp)

### ISSUE THREAD

**Issue by ElektroStudios** [yt-dlp/yt-dlp#3678]
### Checklist

- [X] I'm asking a question and **not** reporting a bug/feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions including closed ones

## I think this could be related with Cloudflare or reCaptcha for some reason.

Yesterday I used **youtube-dl** to download without no problems a entire _non-premium_ _non-login-required_ anime season from **Crunchyroll** website ( this anime: https://www.crunchyroll.com/es-es/hundred ). After that, **youtube-dl** stopped working when I tried to download more videos, giving me HTTP error 403 when I try to download any video from **Crunchyroll**...

So I took a research and investigated all related about cookies usage and I discovered **yt-dlp**, which in theory has the **Chrunchyroll** extractor more updated than **youtube-dl**, so I'm using **yt-dlp** right now.

With **yt-dlp**, on my first try, I only get the same 403 error for any **Crunchyroll** video url like with **youtube-dl**.

Because I can access **Crunchyroll** website with no problem under my local web-browser: **Firefox DevEdition**. 
But yesterday after I finished downloading all the 12 episodes with **youtube-dl**, and it stopped working giving me error 403, I noticed that **Crunchyroll** website asked me to solve a **reCaptcha** in my local **Firefox DevEdition** web-browser, but it happened only once.

If I delete all cookies from my **Firefox DevEdition** browser, and I access **Crynchyroll** website, it asks me to resolve a **reCapctha** only once.

I also noticed that if I use [CefSharp](https://github.com/cefsharp/CefSharp) web-browser in a .NET Windows Forms application, the first time that I try to access **Crunchyroll** website it always ask me to resolve a **reCaptcha**. This happens once every time that I run a new instance of the application. After I solved the **reCaptcha** it does not ask me anymore while the application instance is running. This never happened to me before I downloaded the 12 episodes with youtube-dl.

In my **Firefox DevEdition** web-browser I tried to use [cookies.txt extension](https://addons.mozilla.org/en-US/firefox/addon/cookies-txt/) to generate a cookies file from **Crunchyroll** website. But when I pass these cookies.txt file to **yt-dlp** it does not solve the problem, I still get error 403.

I also tried this alternative syntax to take cookies from my web-browser:

`yt-dlp.exe --cookies-from-browser "firefox:ElektroStudios.dev-edition" --get-filename -o "test video.%(ext)s" "https://www.crunchyroll.com/es-es/hundred/episode-1-little-garden-702479"`

That command execution it just throws error 403 again and again.

I tried other parameters like removing cache dir, check no certificate, setting the same user agent as my Firefox web-browser, and username + password.

I also tried using `--user-agent "Mozilla/5.0"` as suggested [here](https://github.com/ytdl-org/youtube-dl/issues/30926#issuecomment-1120399701). It still gets error 403 with or without cookies.

What can I do to avoid this 403 error and finally let me download videos from **Crunchyroll** again?.

### Cookies.txt file content
```
# Netscape HTTP Cookie File
# https://curl.haxx.se/rfc/cookie_spec.html
# This is a generated file! Do not edit.

www.crunchyroll.com	FALSE	/	FALSE	1652010928	cf_chl_2	9f3a7a1794ff871
www.crunchyroll.com	FALSE	/	FALSE	1652011024	cf_chl_prog	x14
#HttpOnly_.crunchyroll.com	TRUE	/	TRUE	1683547024	cf_clearance	ieG4GNX1pmXKTG6c2LcgKhvEkc8bvWWaisNvaM4zBxg-1652007424-0-250
#HttpOnly_.crunchyroll.com	TRUE	/	FALSE	1667559424	c_visitor	4acfe514-2de9-4994-880f-36ded63f323e
#HttpOnly_.crunchyroll.com	TRUE	/	TRUE	1652009224	__cf_bm	D8vSxRBb4t9zm2X2NUwRnFjNzboYdfAE9Th9U4JVv_Y-1652007424-0-AWaAkapHhy9vfAZgOUnPtG3vf0ieECPCZrGyg/MI4DdOni7GKSaYaMNlmzVTh6XUAVYnKvA4o2dLmj7+loj+d8O2+v4tWi+jm2RTvjNYcxei
www.crunchyroll.com	FALSE	/es-es	FALSE	1686221830	_evidon_consent_cookie	{"consent_date":"2022-05-08T10:57:10.749Z","consent_type":1}
www.crunchyroll.com	FALSE	/es-es	FALSE	1654599430	_evidon_suppress_gdpr_locale_cookie	accepted
#HttpOnly_.crunchyroll.com	TRUE	/	TRUE	0	__cfruid	c91cf6b0af8161647b92804d2eee06e57bac830f-1652007529
#HttpOnly_.crunchyroll.com	TRUE	/	TRUE	0	session_id	f50cc1151f31662aada2cb5a7a6f0819
#HttpOnly_.crunchyroll.com	TRUE	/	TRUE	1653217139	etp_rt	d6746056-7de4-41be-92eb-3525679a704e
www.crunchyroll.com	FALSE	/es-es/hundred	FALSE	1683543540	crunchyroll_beta_hide_banner	1

```

### Verbose log

```
[debug] Command-line config: ['--cookies-from-browser', 'firefox:ElektroStudios.dev-edition', '--verbose', '--get-filename', '-o', 'test video.%(ext)s', 'https://www.crunchyroll.com/es-es/hundred/episode-1-little-garden-702479']
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "C:\Users\Administrador\AppData\Roaming\Mozilla\Firefox\Profiles\ElektroStudios.dev-edition\cookies.sqlite"
[Cookies] Extracted 2724 cookies from firefox
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2022.04.08 [7884ade] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 2021-11-25-git-522f577d7e-full_build-www.gyan.dev (setts), ffprobe 2021-11-25-git-522f577d7e-full_build-www.gyan.dev
[debug] Optional libraries: brotli, certifi, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [crunchyroll] Extracting URL: https://www.crunchyroll.com/es-es/hundred/episode-1-little-garden-702479
[crunchyroll] 702479: Downloading webpage
ERROR: [crunchyroll] 702479: Unable to download webpage: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 641, in extract
  File "yt_dlp\extractor\crunchyroll.py", line 422, in _real_extract
  File "yt_dlp\extractor\crunchyroll.py", line 275, in _download_webpage
  File "yt_dlp\extractor\common.py", line 932, in _download_webpage
  File "yt_dlp\extractor\common.py", line 800, in _download_webpage_handle
  File "yt_dlp\extractor\common.py", line 785, in _request_webpage

  File "yt_dlp\extractor\common.py", line 767, in _request_webpage
  File "yt_dlp\YoutubeDL.py", line 3601, in urlopen
  File "urllib\request.py", line 531, in open
  File "urllib\request.py", line 640, in http_response
  File "urllib\request.py", line 569, in error
  File "urllib\request.py", line 502, in _call_chain
  File "urllib\request.py", line 649, in http_error_default
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

**Comment by dirkf** [comment]
Related: https://github.com/ytdl-org/youtube-dl/issues/30926

**Comment by pukkandan** [comment]
Duplicate of #2052

**Comment by ElektroStudios** [comment]
Now I can say that the **Cloudflare**'s block which only affects **yt-dlp** http requests lasted for about 24 hours.

At this point I only can suggest that users should set interval settings when using **yt-dlp** (and **youtube-dl**) with **Crunchyroll** website, because it seems so severe and it will block you for "excessive" downloading with http error 403 and for a entire day at least.

I'll share a simple batch-script for Windows users to automate url downloads from **Crunchyroll**. The idea around the writing of this script is to prevent the block for excessive downloading by setting a little interval between requests, and a interval of around one minute between consecutive downloads:

```
@ECHO OFF

Set "WebBrowserProfile=firefox:PROFILE_NAME"
Set "SubtitleLangs=esES,esLA,es,enUS,en"
Set "UrlsFile=%CD%\urls.txt"

(
    "%CD%\yt-dlp.exe" ^
    --sleep-requests "2" ^
    --sleep-subtitles "3" ^
    --min-sleep-interval "60" ^
    --max-sleep-interval "120" ^
    --retries "10" ^
    --file-access-retries "10" ^
    --fragment-retries "10" ^
    --abort-on-unavailable-fragment ^
    --abort-on-error ^
    --write-sub ^
    --sub-lang "%SubtitleLangs%" ^
    --sub-format "ass/srt/best" ^
    --windows-filenames ^
    --paths "%CD%\Crunchyroll Downloads" ^
    --output "%%(series)s\%%(season)s\%%(title)s.%%(ext)s" ^
    --format "bestvideo[height<=1080]+bestaudio/best[height<=1080]" ^
    --cookies-from-browser "%WebBrowserProfile%" ^
    --embed-chapters ^
    --batch-file "%UrlsFile%"
) && (
    CLS
) || (
    ECHO:Download failed.
    PAUSE
    EXIT /B 1
)

ECHO:Download completed.
PAUSE
EXIT /B 0
```

I hope this helps.
